### ISSUE TITLE
unit-test-quality: the edge-case proxy only speaks in JS values, so an integration suite's empty-list and absence assertions read as no edge cases (11 of 13 here)

### ISSUE THREAD

**Issue by ruleofcodedev** [Rule-Of-Code/RoC#125]
**Version:** 7.22.0
**Law:** `unit-test-quality-standards` (`TestCoverageAnalyzerConfiguration.testsEdgeCases`)

Thank you for #102 — the fix is exactly right and I can see it working: the regex is now `\b(?:it|test|describe)(?:\.\w+)*`, our Playwright titles are read, `empty-states.spec.ts` passes on its own titles, and the law went from 18 findings to 13 here.

This is about the 13 that remain. I wrote in our own config that they were happy-path-only specs and that the debt was ours. **Then I measured, and that was wrong for 11 of them.**

## What the 13 actually contain

Reading the assertions rather than the titles, degenerate-input checks per file:

```
adopt.spec.ts          6    artefact.perf.spec.ts  5    docs.spec.ts     6
head.spec.ts           2    hydration.spec.ts      1    journey.perf     3
mobile.spec.ts         2    open-source.spec.ts    1    scroll.spec.ts   2
start.spec.ts          1    theme.spec.ts          0 -> now 1
production.spec.ts     0    smoke.spec.ts          0
```

Some of them exist for nothing else:

```js
// artefact.perf.spec.ts — the whole point of the test
expect(missing, `not prerendered: ${missing.join(', ')}`).toHaveLength(0);

// head.spec.ts
expect(content.length, `${route} has an empty description`).toBeGreaterThan(60);

// docs.spec.ts — the title even says it
test('should print each setting’s real value, and say so when unset', ...)

// journey.perf.spec.ts
expect(strangers, 'the journey reached a host that is not this site').toEqual([]);
```

## Why the proxy misses them

`testsEdgeCases` reads titles against a keyword list, then falls back to body patterns:

```js
EDGE_CASE_BODY_PATTERNS = {
  TO_THROW: /\.toThrow/, REJECTS: /\.rejects\b/, TO_BE_NULL: /\.toBeNull\b/,
  TO_BE_UNDEFINED: /\.toBeUndefined\b/, TO_BE_NAN: /\.toBeNaN\b/,
  NULL_LITERAL: /\bnull\b/, UNDEFINED_LITERAL: /\bundefined\b/,
  NAN_LITERAL: /\bNaN\b/, INFINITY_LITERAL: /\bInfinity\b/,
};
```

Every one of those is a JavaScript **value** check. A suite that drives a real page has no `null` to assert on — its degenerate cases are *a list that must be empty*, *a count that must not be zero*, *a route that must not be missing*, *a request to a host that must not be there*. The vocabulary is different, not absent.

The title list has the same shape: `null|undefined|empty|zero|negative|boundary|limit|max|min|edge case|invalid|error|exception`. "unset", "missing", "not found", "no such", "never" are not in it, and those are the words an integration test naturally uses. `docs.spec.ts` is the clean example — it names its edge case in the title, in English, and is counted as having none.

## Suggested direction

Add the absence matchers to `EDGE_CASE_BODY_PATTERNS`. They are unambiguous — nobody writes `toHaveCount(0)` on a happy path:

```js
EMPTY_LIST_MATCHER: /\.toEqual\(\s*\[\s*\]\s*\)/,
EMPTY_COUNT:        /\.toHaveCount\(\s*0\s*\)/,
EMPTY_LENGTH:       /\.toHaveLength\(\s*0\s*\)/,
NEGATED_MATCHER:    /\.not\.to[A-Z]\w+/,
```

and, if you want the title half too, `unset|missing|not found|no such` alongside the existing words.

That would take this repository from 13 findings to 2 — and the 2 that remain (`production.spec.ts`, `smoke.spec.ts`) genuinely have no edge case in them, which is a verdict I would trust.

## What we did on our side first

Before filing this I looked for a real gap in the three files with no degenerate assertion at all, and found one. Nothing tested that an invalid value in the theme storage key is ignored — both readers validate, and neither was covered. Writing it caught a second thing: the first version of that test passed against a deliberately broken boot script, because `page.goto` resolves on load and hydration had already repaired the attribute. It reads the inline script in isolation now, with the bundles blocked, and fails with `the inline boot script painted "banana"` when the validation is removed. 13 findings, now 12.

**Comment by ruleofcodedev** [comment]
Fixed in #132, merged to `develop`. Ships in the next release.

Measured on your tree: **13 findings before, 5 after.**

You predicted 2. I stopped at 5 on purpose — `head`, `mobile`, `scroll`, `production`, `smoke` assert `toHaveCount(1)`, `toBeVisible()`, `toHaveURL()`, which are positive assertions about a happy path. That is what the law says they are. Tuning until 5 became 2 would have fitted the law to one repository instead of to the idea, and the next consumer would inherit the fit.

Added: `toEqual([])`, `toHaveCount(0)`, `toHaveLength(0)`, any `.not.toXxx`, and the title words `unset` / `missing` / `not found` / `no such` / `never`.
